### DISCUSSION TITLE
Random Forest Algorithm Performance

### DISCUSSION THREAD

**Discussion by ToniAS21** [teamalgoritma/community#332] [Q&A] [answered]
Permisi rekan-rekan. Saya mau nanya terkait project Capstone Machine Learning Airlines13.

Saya coba berbagai algoritma untuk menyelesaikan projek tersebut dan mendapatkan hasil yang lumayan lebih baik di algoritma random forest. Nah saya bingggung pas saya tuning untuk trainControlnya (number dan repeatnya) tapi hasil peningkatan performanya tidak signifikan atau dapat dikatakan konstan meskipun saya ubah number dan repeatnya lebih tinggi. Kira-kira disebabkan oleh apa ya? kemudian bagaimana tips meningkatkan performa model random forest.

Sebelumnya saya sudah melakukan preprocessing data (mengatasi missing value, menghapus kolom yang tidak diperlukan dsb), penyeimbangan proporsi kelas train dsb.

Berikut code pas membangun model

![Screenshot (466)](https://user-images.githubusercontent.com/83503249/191786924-2ed23c9d-11f1-4180-81fb-eafac801f46a.png)

Berikut hasil ketika tuning.

**1. number = 2, repeat = 1**
![Screenshot (464)](https://user-images.githubusercontent.com/83503249/191786418-19964d35-f896-43fa-8304-f5790576dc7f.png)

**2. number = 5, repeat = 1**
![Screenshot (462)](https://user-images.githubusercontent.com/83503249/191787434-d13ff47b-b09a-4fde-bb6a-46c661b08890.png)

**3. number = 5, repeat = 2**
![Screenshot (467)](https://user-images.githubusercontent.com/83503249/191788384-471a3608-285a-4452-8ed2-d3b46b762c92.png)

**4. number = 10, repeat = 3**
![Screenshot (465)](https://user-images.githubusercontent.com/83503249/191787717-618716c6-6388-48c7-8bc2-597a6398fe1f.png)

**3. number = 10, repeat = 5**
![Screenshot (461)](https://user-images.githubusercontent.com/83503249/191788577-3725930d-c602-45e0-a0be-053364d88ad6.png)

**Comment by NabiilahArdini** [comment, marked as answer]
Dear Bapak @ToniAS21, apabila peningkatan `number` dan `repeats` tidak memberikan peningkatan performa tidak signifikan, maka kemungkinan model sudah convergen pada `number` dan `repeats` yang digunakan di awal. Bila kasusnya seperti itu, untuk meningkatkan performa, dapat melalui tahap pre-processing data, misalnya scaling data, handling NA, feature extraction, dll. 
Demikian yang dapat saya sarankan. Semoga dapat bermanfaat!

**Reply by ToniAS21** [reply]
Baik terima kasih kak @NabiilahArdini
